public/lib/general.js

```diff
@@ -1,121 +1,136 @@
-(function() {
-	"use strict";
-	
-	var articlePath = window.location.protocol + '//' + window.location.host + window.location.pathname;
+'use strict';
 
-	var pluginURL = nodeBBURL + '/plugins/nodebb-plugin-blog-comments',
-		savedText, nodebbDiv, contentDiv, commentsDiv, commentsCounter, commentsAuthor, commentsCategory;
+/* eslint-disable no-undef */
+(function () {
+	const articlePath = `${window.location.protocol}//${window.location.host}${window.location.pathname}`;
 
-	var stylesheet = document.createElement("link");
-	stylesheet.setAttribute("rel", "stylesheet");
-	stylesheet.setAttribute("type", "text/css");
-	stylesheet.setAttribute("href", pluginURL + '/css/comments.css');
+	const pluginURL = `${nodeBBURL}/plugins/nodebb-plugin-blog-comments`;
+	let savedText; let contentDiv; let commentsDiv; let commentsCounter; let commentsAuthor; let
+		commentsCategory;
 
-	document.getElementsByTagName("head")[0].appendChild(stylesheet);
+	const stylesheet = document.createElement('link');
+	stylesheet.setAttribute('rel', 'stylesheet');
+	stylesheet.setAttribute('type', 'text/css');
+	stylesheet.setAttribute('href', `${pluginURL}/css/comments.css`);
+
+	document.getElementsByTagName('head')[0].appendChild(stylesheet);
 	document.getElementById('nodebb-comments').insertAdjacentHTML('beforebegin', '<div id="nodebb"></div>');
-	nodebbDiv = document.getElementById('nodebb');
+	const nodebbDiv = document.getElementById('nodebb');
 
 	function newXHR() {
 		try {
-	        return XHR = new XMLHttpRequest();
-	    } catch (e) {
-	        try {
-	            return XHR = new ActiveXObject("Microsoft.XMLHTTP");
-	        } catch (e) {
-	            return XHR = new ActiveXObject("Msxml2.XMLHTTP");
-	        }
-	    }
+			return new XMLHttpRequest();
+		} catch (e) {
+			try {
+				return new ActiveXObject('Microsoft.XMLHTTP');
+			} catch (e) {
+				return new ActiveXObject('Msxml2.XMLHTTP');
+			}
+		}
 	}
 
-	var XHR = newXHR(), pagination = 0, modal;
-
-	function authenticate(type) {
-		savedText = contentDiv.value;
-		modal = window.open(nodeBBURL + "/" + type + "/#blog/authenticate","_blank","toolbar=no, scrollbars=no, resizable=no, width=600, height=675");
-		var timer = setInterval(function() {
-			if(modal.closed) {  
-				clearInterval(timer);
-				pagination = 0;
-				reloadComments();
-			}  
-		}, 500);
-	}
+	const XHR = newXHR(); let pagination = 0;
```

Comment:
Put these on separate lines. We should add https://github.com/nodebb/eslint-config-nodebb to align with other plugins.